### DISCUSSION TITLE
Azure PUE and percentage renewable energy for some regions

### DISCUSSION THREAD

**Discussion by autosysops** [cloud-carbon-footprint/cloud-carbon-footprint#1299] [Feature suggestions]
Microsoft provided a sustainability fact sheet for some regions here: https://datacenters.microsoft.com/globe/
I've written a script which parses the provided pdf's and extracts the data from it and presents it in a JSON which can be found here:
https://raw.githubusercontent.com/autosysops/azure_sustainability_data/main/regiondata.json
You can also find the whole repository here:
https://github.com/autosysops/azure_sustainability_data
It will scan every day and automatically create a pull request if any new data is available, so after a manual check it will be added to the file.
I thought that might help as for now the PUE and percentage of renewable energy for the whole of Azure is the same.

Microsoft doesn't provide this information for every region yet, so this list only provides the data for the regions where they do provide the information, for the other regions the current avarage should probably still be used.

I hope this could help and if I need to change it in any way to make it even better let me know.

**Comment by 4upz** [comment]
@autosysops this is awesome! Thanks for creating this. We think this script could be extremely valuable since we don't have renewable energy percentages for Azure when using the default carbon intensity values, nor do we have a way of automating this data. 

Would you be interested in contributing this type of work to our [Coefficients Repo](https://github.com/cloud-carbon-footprint/ccf-coefficients)? We do similar work where we track and automate the updates for our machine coefficients. We also hold the regional carbon intensity data that CCF uses here as well. The caveat is that the repository is in Python and I see your script is in Powershell, but happy to support you in making the changes needed to help it fit into the rest of the codebase.

**Reply by autosysops** [reply]
I'm interested in helping out. But for the first two/three weeks I wont have any time. I used to do some python but it has been a while so I would need to check up on it again. The hardest part would be to parse the PDF files as I use a dot net library to parse these which gives me the text in a kind of formatted way, so I don't know if something similar exists for python.

**Reply by 4upz** [reply]
@autosysops Not a problem, we're happy to help collab on this. I'm not the biggest Python expert myself, but I'm sure there's a library out there for PDF parsing. Worst-case, we can see if there's an alternative source for grabbing the data.

Let's pick this up when you have time, as I would love for your work to be used! Maybe shoot me a message/email in 2-3 weeks when your schedule calms down?
